services/repository/contributors_graph.go

```diff
@@ -199,6 +220,31 @@ func getExtendedCommitStats(repo *git.Repository, revision string /*, limit int
 	return extendedCommitStats, nil
 }
 
+var errSyntax = errors.New("syntax error occurred")
+
+func parseCoAuthorTrailerValue(value string) (email, name string, err error) {
+	value = strings.TrimSpace(value)
+	if !strings.HasSuffix(value, ">") {
+		return "", "", errSyntax
+	}
+	if openEmailBracketIdx := strings.LastIndex(value, "<"); openEmailBracketIdx == -1 {
+		return "", "", errSyntax
+	}
+
+	parts := strings.Split(value, "<")
+	if len(parts) < 2 {
+		return "", "", errSyntax
+	}
+
+	email = strings.TrimRight(parts[1], ">")
+	if _, err := mail.ParseAddress(email); err != nil {
```

Comment:
Since users can add anything between the <> brackets I believed that we could try to throw away lines that contain invalid email addresses.

Though while testing by creating blank repos and merging the commit history of existing projects into these repos I observed that some emails associated to bots will be rejected (e.g. the pre-commit bot on GitHub). I guess this happens because these emails don't follow [RFC 5322](https://pkg.go.dev/net/mail#ParseAddress). Not sure if I should do this validation check.

Reply:
Depends on how we parse other trailer lines I suppose. I would make it consistent with those.